packages/runed/src/lib/utilities/element-size/element-size.svelte.ts

```diff
@@ -79,14 +79,56 @@ export class ElementSize {
 		this.#node = node;
 
 		this.#size = {
-			width: options.initialSize?.width ?? 0,
-			height: options.initialSize?.height ?? 0,
+			width: 0,
+			height: 0,
 		};
 	}
 
+	calculateSize() {
+		const element = get(this.#node);
+
+		// no element or no window, return undefined, we will return the initial size
+		if (!element || !this.#window) {
+			return;
+		}
+
+		const offsetWidth = element.offsetWidth;
+		const offsetHeight = element.offsetHeight;
+
+		// easy mode, just return offsets
+		if (this.#options.box === "border-box") {
+			return {
+				width: offsetWidth,
+				height: offsetHeight,
+			};
+		}
+
+		// hard mode, we need to calculate the content size
+		const style = this.#window.getComputedStyle(element);
+
+		const paddingWidth = parseFloat(style.paddingLeft) + parseFloat(style.paddingRight);
+		const paddingHeight = parseFloat(style.paddingTop) + parseFloat(style.paddingBottom);
+		const borderWidth = parseFloat(style.borderLeftWidth) + parseFloat(style.borderRightWidth);
+		const borderHeight = parseFloat(style.borderTopWidth) + parseFloat(style.borderBottomWidth);
+
+		const contentWidth = offsetWidth - paddingWidth - borderWidth;
+		const contentHeight = offsetHeight - paddingHeight - borderHeight;
+
+		return {
+			width: contentWidth,
+			height: contentHeight,
+		};
+	}
+
+	getSize() {
+		// if the resize observer already run we can just return the size
+		// otherwise we calculate the size if possible or we return the initial size
+		return this.#updated ? this.#size : (this.calculateSize() ?? this.#size);
```

Comment:
it would be nice if we didn't need to do this ternary

i.e. once, when we initialise, `calculateSize` (because the resize observer can't possibly have run yet)

 all other cases involve the observer running and updating the value from then on

Reply:
If you don't read the value in a tracking context createSubscriber will not run...so you could read the value outside of tracking context, change the size and then read the value again...we need to recalculate everytime.

Reply:
would the ternary only ever hit the falsy condition once though?

`#updated` will only be `false` once? that's what irks me. i wonder if we can do the initialisation more explicitly rather than checking every time. then this just always returns `#size`

Reply:
`#updated` is set back to false on the cleanup of the `createSubscriber` and that can happen mid application if you clean up every effect that was reading the value (for example if you are reading the value in an if and the condition changes).

But even then if you do something like

```ts
const size = new ElementSize(div);
console.log(size.width);
div.style = "width: 1000px";
console.log(size.width);
```
in a non reactive context the second width should be 1000 not the original one no?

Reply:
yeah i understand now 🤔 

you would expect it to be `1000`

Reply:
I mean hopefully people will actually use this in reactive contexts for elements that can change in size so it shouldn't be too much of a problem 😄